docs/backup-recovery/README.md

```diff
@@ -0,0 +1,146 @@
+# Cluster upgrades - backup and recovery #
+
+## Background ##
+
+This feature creates a pre-upgrade backup and provides a procedure for rapid recovery of a SNO in the event of a failed upgrade. In case of an upgrade failure, this feature allows the SNO to be restored to a working state with the previous version of OCP without requiring a re-provision of the application(s).  
+
+## Create Recovery Partition ##
+
+> **WARNING**: It is highly recommended to create a recovery partition at install time if opted to use this feature.
+
+There are two methods related to partition creation is described here:
+
+### Preferred method with Siteconfig ###
+
+The recovery partition can be created at install time by defining a node level property called `diskPartition` in the SiteConfig. The configuration for the partition must be defined by `start` parameter which must indicate the end of the root device. Additionally, a spare disk can be used for recovery partition as well. A recovery partition of 50GB can be created by following the below example:
+
+```
+nodes:
+      - hostName: "snonode.sno-worker-0.e2e.bos.redhat.com"
+        role: "master"
+        rootDeviceHints:
+          hctl: "0:2:0:0"
+          deviceName: /dev/sda
+        ........
+        ........
+        #Disk /dev/sda: 893.3 GiB, 959119884288 bytes, 1873281024 sectors
+        diskPartition:
+          - device: /dev/sda
+            partitions:
+              - mount_point: /var/recovery
+                size: 51200
+                start: 800000
+
+```
+
```

Comment:
The result of applying this configuration is the following filesystem layout. See that the first 4 partitions are created on any RHCOS installation. The 5th partition is the extra partition that we requested to store all backup and recovery files. It is mounted in the /var/recovery folder at boot time using a systemd mount service created automatically by ZTP. 

```
NAME    MAJ:MIN RM   SIZE RO TYPE MOUNTPOINT
sda       8:0    0 893.3G  0 disk
├─sda1    8:1    0     1M  0 part
├─sda2    8:2    0   127M  0 part
├─sda3    8:3    0   384M  0 part /boot
├─sda4    8:4    0 733.1G  0 part /sysroot
└─sda5    8:5    0 50G  0 part /var/recovery
```

Reply:
As discussed today at the platform lifecycle meeting with @donpenney  and others, we would like to scope the readme with associated development done in the repo. It seems this detail really enhances the understanding of the partition creation but not a feature of this repo rather from ZTP. I would like to add this part to the design doc so that Doc team can use this info for their higher level doc for the client.